concoction-lib/src/main/java/info/mmpa/concoction/scan/model/dynamic/DynamicMatchingModel.java

```diff
@@ -37,41 +37,33 @@ public DynamicMatchingModel(@Nonnull Map<String, DynamicMatchEntry> variants) {
 	 * 		Sink to feed match results into.
 	 * @param archetype
 	 * 		Information about what the signature being matched.
-	 * @param sourcePath
+	 * @param path
 	 * 		Current method path to the containing input source. SSVM holds the rest of the details.
 	 * @param frame
 	 * 		SSVM frame of method entered.
 	 */
 	public void matchOnEnter(@Nonnull ResultsSink sink, @Nonnull DetectionArchetype archetype,
-							 @Nonnull SourcePathElement sourcePath, @Nonnull CallStackFrame frame) {
+							 @Nonnull MethodPathElement path, @Nonnull CallStackFrame frame) {
 		for (DynamicMatchEntry entry : variants.values())
-			if (entry.matchOnEnter(frame)) {
-				MethodPathElement path = sourcePath
-						.child(frame.getOwnerName())
-						.child(frame.getMethodName(), frame.getMethodDesc());
-				sink.add(path, archetype, new Detection(archetype, path));
-			}
+			if (entry.matchOnEnter(frame))
+				Unchecked.runSafe("dynamic-sink-feed", () -> sink.onDetection(path, archetype, new Detection(archetype, path)));
 	}
 
 	/**
 	 * @param sink
 	 * 		Sink to feed match results into.
 	 * @param archetype
 	 * 		Information about what the signature being matched.
-	 * @param sourcePath
+	 * @param path
 	 * 		Current method path to the containing input source. SSVM holds the rest of the details.
 	 * @param frame
 	 * 		SSVM frame of method exited.
 	 */
 	public void matchOnExit(@Nonnull ResultsSink sink, @Nonnull DetectionArchetype archetype,
-							@Nonnull SourcePathElement sourcePath, @Nonnull CallStackFrame frame) {
+							@Nonnull MethodPathElement path, @Nonnull CallStackFrame frame) {
```

Comment:
Do we want to change this? Seems less descriptive to me.

Reply:
The type is changed. This should only ever consume a method path.

Reply:
Could we change it to `methodPath` then?

Reply:
Done